app/src/main/java/com/example/lutsak/fragments/MainActivity.java

```diff
@@ -1,13 +1,74 @@
 package com.example.lutsak.fragments;
-
-import android.support.v7.app.AppCompatActivity;
+import android.app.Activity;
+import android.app.FragmentTransaction;
 import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
 
-public class MainActivity extends AppCompatActivity {
+public class MainActivity extends Activity implements View.OnClickListener{
 
-    @Override
-    protected void onCreate(Bundle savedInstanceState) {
+    final String LOG_TAG = MainActivity.class.getSimpleName();
+
+    private Fragment1 mFragment1;
```

Comment:
You simply cannot keep a reference to the fragment in this way.
You can fetch them from getSupportFragmentManager().findFragmentByTag(...)

These 2 references will be destroyed when activity gets destroyed.
Try enabling Don't keep activities and put application to background and then back to foreground